python/dash_tools/mp4.py

```diff
@@ -837,20 +835,36 @@ def childpos(self):
         return self.offset+36
 
 
-class dac3_box(SampleEntry):
+class dac3_box(box):
     def __init__(self, *args):
-        SampleEntry.__init__(self, *args)
+        box.__init__(self, *args)
         self.dec_info = self.fmap[self.offset+8:self.offset+self.size]
         self.dec_info_hex = ''.join(['%02x' % ord(c) for c in self.dec_info])
         self.decoration = 'dec_info={0}'.format(self.dec_info_hex)
 
 
-class dec3_box(SampleEntry):
+class dec3_box(box):
     def __init__(self, *args):
-        SampleEntry.__init__(self, *args)
+        box.__init__(self, *args)
         self.dec_info = self.fmap[self.offset+8:self.offset+self.size]
         self.dec_info_hex = ''.join(['%02x' % ord(c) for c in self.dec_info])
-        self.decoration = 'dec_info={0}'.format(self.dec_info_hex)
+
+        # https://www.etsi.org/deliver/etsi_ts/102300_102399/102366/01.03.01_60/ts_102366v010301p.pdf
+        # EC3SpecificBox F.6.1 page 202
+        self.data_rate = (ord(self.dec_info[0]) << 5) + ((ord(self.dec_info[1]) >> 3) & 0x1f)
+        self.num_ind_sub = ord(self.dec_info[1]) & 0x7
+
+        # NOTE: currently only parsing the first dependent substream
+        self.fscod = (ord(self.dec_info[2]) >> 6) & 0x3
+        bsid = (ord(self.dec_info[2]) >> 1) & 0x1f
+        asvc = (ord(self.dec_info[3]) >> 7) & 0x1
+        bsmod = (ord(self.dec_info[3]) >> 4) & 0x7
+        self.acmod = (ord(self.dec_info[3]) >> 1) & 0x7
+        lfeon = ord(self.dec_info[3]) & 0x1
+        num_dep_sub = (ord(self.dec_info[4]) >> 1) & 0xf
+
```

Comment:
I think it would be good to parse the last bit in the case num_dep_sub == 0, and 9 bits for chan_loc if num_dep_sub > 0.

Reply:
Good point, I have added the suggested parsing.